compiler/infra/InterferenceGraph.cpp

```diff
@@ -406,7 +405,12 @@ bool TR_InterferenceGraph::simplify()
          while (bvi.hasMoreElements())
             {
             igNode = getNodeTable(bvi.getNextElement());
-            if (igNode->getDegree() > degree)
+
+            // TODO: This unsigned conversion was inserted while addressing warnings. The warning at this line warned
+            // us about signed/unsigned comparison. The C++ standard dictates that signed values are always converted
+            // to unsigned before the comparison. Because `degree` is initialized to -1, the unsigned conversion will
+            // produce UINT_MAX in this case, so the comparison below will never succeed and we should always assert.
+            if (igNode->getDegree() > unsigned(degree))
```

Comment:
Highlighting this bug as well. I'm failure sure this code is dead though as we haven't done graph colouring in a while.

Reply:
Actually, I believe graph colouring is still used for compacting locals, though it's unclear how much of the code actually gets used:

https://github.com/eclipse/omr/blob/e358cd0e50776aacc01d109deb37c454901dad05/compiler/codegen/CodeGenPrep.cpp#L352-L416